cheatsheets/Cryptographic_Storage_Cheat_Sheet.md

```diff
@@ -1,234 +1,185 @@
+- Generating and storing new keys.
+- Distributing keys to the required parties.
+- Deploying keys to application servers.
+- Rotating and decommissioning old keys
 
-##### 3.5 Protect any keys used to secure cardholder data against disclosure and misuse
+### Key Generation
 
-As the requirement name above indicates, we are required to securely store the encryption keys themselves. This will mean implementing strong access control, auditing and logging for your keys. The keys must be stored in a location which is both secure and "away" from the encrypted data. This means key data shouldn't be stored on web servers, database servers etc
+Keys should be randomly generated using a cryptographically secure function, such as those discussed in the [Secure Random Number Generation](#secure-random-number-generation) section. Keys **should not** be based on common words or phrases, or on "random" characters generated by mashing the keyboard.
 
-Access to the keys must be restricted to the smallest amount of users possible. This group of users will ideally be users who are highly trusted and trained to perform Key Custodian duties. There will obviously be a requirement for system/service accounts to access the key data to perform encryption/decryption of data.
+Where multiple keys are used (such as data separate data-encrypting and key-encrypting keys), they should be fully independent from each other.
 
-The keys themselves shouldn't be stored in the clear but encrypted with a KEK (Key Encrypting Key). The KEK must not be stored in the same location as the encryption keys it is encrypting.
+### Key Lifetimes and Rotation
 
-##### 3.6 Fully document and implement all key-management processes and procedures for cryptographic keys used for encryption of cardholder data
+Encryption keys should be changed (or rotated) based on a number of different criteria:
 
-Requirement 3.6 mandates that key management processes within a PCI compliant company cover 8 specific key lifecycle steps:
+- If the previous key is known (or suspected) to have been compromised.
+  * This could also be caused by a someone who had access to the key leaving the organisation.
+- After a specified period of time has elapsed (known as the cryptoperiod).
+  * There are many factors that could affect what an appropriate cryptoperiod is, including the size of the key, the sensitivity of the data, and the threat model of the system. See section 5.3 of [NIST SP 800-57](https://nvlpubs.nist.gov/nistpubs/SpecialPublications/NIST.SP.800-57pt1r4.pdf) for further guidance.
+- After the key has been used to encrypt a specific amount of data.
+  * This would typically be `2^35` bytes (~34GB) for 64-bit keys (DES, 3DES, Blowfish, RC5, etc.) and `2^68` bytes (~295 exabytes) for 128 bit keys (AES, Twofish, Serpent, etc.).
```

Comment:
DES has a 64bit block size but not a 64bit key size, it has a 56bit key size. 3DES has either a 128bit or 192bit key size. This area needs review.

Reply:
Good call - this was from the original CS and has some issues. I've just removed the algorithms in 220d6ee - the message is just as clear but without the list of (somewhat incorrect) examples cluttering it up.